src/Window.cc

```diff
@@ -517,37 +520,21 @@ void FluxboxWindow::init() {
             // this probably should never happen, but if a window
             // unexplicitly has its topleft corner outside any screen,
             // move it to the current screen and ensure it's just placed
+            // this typically happens with fullscreen+ windows, e.g. videos
             int cur = screen().getHead(fbWindow());
-            move(screen().getHeadX(cur), screen().getHeadY(cur));
-            m_placed = false; // allow placement strategy to fix position
+            int new_x = screen().getHeadX(cur), new_y = screen().getHeadY(cur);
+            if (m_client->height() >= screen().height())
+              new_x = 0;
```

Comment:
Also hardcoded 0/0 might cause trouble if the area isn't covered by a multiscreen layout and I'm at hand not sure that this won't collide w/ a toolbar on top of the workspace?

Reply:
That makes sensee. Don't have multiscreen layout, only workspaces. What would be the appropriate `new_x`?

New 'screen sized' window is below titlebar when titlebar layer is top and above titlebar when titlebar layer is normal. Fullscreen window goes over titlebar.

Reply:
I'm curious to udnerstand the specific usecase here.

The specific result will be a window w/ a titlebar possibly below the toolbar and a portion of the video off-screen.
Is that actually desired?

Let's pretend you're playing an 8k video on a 1080p output, that has 90+% of the video off-screen.
Do you really think this is what you or anyone else wants?

What if the client screws up it's config and tries to create a maximum size window (iirc 15 or 16 bits per dimension) and burns through your RAM on the way there. Is that what you want?

I can reason to maximize the window to the available output or to fullscreen it, but allowing windows - generally - to exceed the screen estate and move parts of them off screen (when the windows signals that it's fine w/ other dimensions) seems pointless?

Reply:
This puts an oversized window on the top left of the screen, not below the toolbar's XY, below (or over) the toolbar's **layer**.

Resizing oversized windows makes sense. The maximized I don't expect to get because no window starts maximized by default. Starting fullscreen is similar to maximized but it will be over the toolbar's **layer**.

Reply:
> This puts an oversized window on the top left of the screen, not below the toolbar's XY, below (or over) the toolbar's layer.
I've a hard time parsing that sentence but in any event, I'm not asking what the patch is inteded to to but what the general idea is.

The situation we're facing is a window that exceeds the screen size, afaiu you initiall (on 1.3.7) faced a weird layering bug but w/ the git master the window gets maximized to the workspace.
You also seem to agree with shrinking the window to the available workspace but take issue w/ setting the window maximized?
Why?
The only time this is relevant (?) is when you click the maximization button which will presently unmaximize the window - otherwise it would toggle the maximized state but not respond otherwise (as the maximum size is already taken)
I'm not sure how this would be beneficial?

Reply:
I just don't expect a window to start maximized, it just looks like a big window. I have not seen that behavior anywhere before. Granted I've been running Fluxbox for years so not sure if it happens elsewhere.

Running without the 'maximize' lines (below) seems to start the window 2/3 screen size which feels more natural.

**master** will create a maximized window, going to fullscreen goes to fullscreen, exiting fullscreen goes to maximized window which confused

**master+setFullscreen patch** will create a fullscreen window, exiting fullscreen the titlebar is above the screen space XY

**master+remove maximize** creates a non-maximized 2/3 screen size window, no weirdness imo

**my last commit** was doing a window move so the titlebar is still visible on the screen though as you mentioned I set new_x/new_y wrong.

After looking at the non-max'd 2/3 window I think makes sense and feels the best.

```diff
--- a/src/Window.cc  2022-04-18 15:59:33.000000000 -0400
+++ b/src/Window.cc  2022-05-02 09:14:10.092678272 -0400
@@ -532,19 +532,19 @@
     if (!m_state.fullscreen) {
         unsigned int new_width = 0, new_height = 0;
         if (m_client->width() >= screen().width()) {
-            m_state.maximized |= WindowState::MAX_HORZ;
+            //m_state.maximized |= WindowState::MAX_HORZ;
             new_width = 2 * screen().width() / 3;
         }
         if (m_client->height() >= screen().height()) {
-            m_state.maximized |= WindowState::MAX_VERT;
+            //m_state.maximized |= WindowState::MAX_VERT;
             new_height = 2 * screen().height() / 3;
         }
         if (new_width || new_height) {
-            const int maximized = m_state.maximized;
-            m_state.maximized = WindowState::MAX_NONE;
+            //const int maximized = m_state.maximized;
+            //m_state.maximized = WindowState::MAX_NONE;
             resize(new_width ? new_width : width(), new_height ? new_height : height());
             m_placed = false;
-            m_state.maximized = maximized;
+            //m_state.maximized = maximized;
         }
     }
```